integrations/cohere/src/haystack_integrations/components/embedders/cohere/utils.py

```diff
@@ -24,20 +24,14 @@ async def get_async_response(cohere_async_client: AsyncClient, texts: List[str],
     """
     all_embeddings: List[List[float]] = []
     metadata: Dict[str, Any] = {}
-    try:
-        response = await cohere_async_client.embed(
-            texts=texts, model=model_name, input_type=input_type, truncate=truncate
-        )
-        if response.meta is not None:
-            metadata = response.meta
-        for emb in response.embeddings:
-            all_embeddings.append(emb)
 
-        return all_embeddings, metadata
+    response = await cohere_async_client.embed(texts=texts, model=model_name, input_type=input_type, truncate=truncate)
+    if response.meta is not None:
+        metadata = response.meta
+    for emb in response.embeddings:
+        all_embeddings.append(emb)
 
-    except CohereError as error_response:
-        msg = error_response.message
-        raise ValueError(msg) from error_response
+    return all_embeddings, metadata
```

Comment:
I've removed the `try / except` block around this code entirely since the SDK no longer exports `CohereError`. Is there a reason for trapping exceptions from the SDK and converting them to `ValueError`s, or is it OK to let those exceptions bubble up?

Reply:
The .embed code can [raise a couple of different exceptions:](https://github.com/cohere-ai/cohere-python/blob/main/src/cohere/base_client.py#L2649) BadRequestError, TooManyRequestsError, InternalServerError, JSONDecodeError, ApiError. They are all not ValueErrors but inherit from ApiError and Exception: https://github.com/cohere-ai/cohere-python/blob/437fdc65e969bf6e69c2639ae906671912436c5f/src/cohere/core/api_error.py#L6
An application built with this Haystack integration now needs to look out for other types of exceptions but this breaking change is the only disadvantage I see. Your change makes the raised exceptions more informative, which is helpful. It wasn't an actual ValueError before.